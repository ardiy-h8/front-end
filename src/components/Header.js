import React, { Component } from 'react'
import { withStyles } from 'material-ui/styles'
import { AppBar, Toolbar, Typography } from 'material-ui/'

class Header extends Component {
  render () {
    return (
      <div style={styles.root}>
        <AppBar position='static'>
          <Toolbar>
            <Typography variant='title' color='inherit'>
              <img src='./header-logo.png' height='40px' width='40px' />
            </Typography>
          </Toolbar>
        </AppBar>
      </div>
    )
  }
}

const styles = {
  root: {
    width: '100%',
    top: 0,
    zIndex: 100,
    position: 'fixed',
    backgroundColor: '#000080'
  }
}

export default Header
