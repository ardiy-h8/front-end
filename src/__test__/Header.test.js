import React from 'react'
import { AppBar, Toolbar, Typography } from 'material-ui/'


import Header from '../components/Header'
import './__setupTest__.js'



const styles = {
  root: {
    width: '100%',
    top: 0,
    zIndex: 100,
    position: 'fixed',
    backgroundColor: '#000080'
  }
}

describe('Header component', () => {
  it('should be defined', () => {
    expect(Header).toBeDefined()
  })

  it('should render Header component and match Snapshot', () => {
    const wrapper = shallow(
      <div style={styles.root}>
        <AppBar position='static'>
          <Toolbar>
            <Typography variant='title' color='inherit'>
              <img src='./header-logo.png' alt='logo' height='40px' width='40px' />
            </Typography>
          </Toolbar>
        </AppBar>
      </div>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
