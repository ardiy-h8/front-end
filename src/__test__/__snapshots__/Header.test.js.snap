// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header component and match Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    style={
        Object {
            "backgroundColor": "#000080",
            "position": "fixed",
            "top": 0,
            "width": "100%",
            "zIndex": 100,
          }
    }
>
    <WithStyles(AppBar)
        position="static"
    >
        <WithStyles(Toolbar)>
            <WithStyles(Typography)
                color="inherit"
                variant="title"
            >
                <img
                    alt="logo"
                    height="40px"
                    src="./header-logo.png"
                    width="40px"
                />
            </WithStyles(Typography)>
        </WithStyles(Toolbar)>
    </WithStyles(AppBar)>
</div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(AppBar)
        position="static"
>
        <WithStyles(Toolbar)>
                <WithStyles(Typography)
                        color="inherit"
                        variant="title"
                >
                        <img
                                alt="logo"
                                height="40px"
                                src="./header-logo.png"
                                width="40px"
                        />
                </WithStyles(Typography)>
        </WithStyles(Toolbar)>
</WithStyles(AppBar)>,
      "style": Object {
        "backgroundColor": "#000080",
        "position": "fixed",
        "top": 0,
        "width": "100%",
        "zIndex": 100,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Toolbar)>
          <WithStyles(Typography)
                    color="inherit"
                    variant="title"
          >
                    <img
                              alt="logo"
                              height="40px"
                              src="./header-logo.png"
                              width="40px"
                    />
          </WithStyles(Typography)>
</WithStyles(Toolbar)>,
        "position": "static",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Typography)
            color="inherit"
            variant="title"
>
            <img
                        alt="logo"
                        height="40px"
                        src="./header-logo.png"
                        width="40px"
            />
</WithStyles(Typography)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              alt="logo"
              height="40px"
              src="./header-logo.png"
              width="40px"
/>,
            "color": "inherit",
            "variant": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "height": "40px",
              "src": "./header-logo.png",
              "width": "40px",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(AppBar)
          position="static"
>
          <WithStyles(Toolbar)>
                    <WithStyles(Typography)
                              color="inherit"
                              variant="title"
                    >
                              <img
                                        alt="logo"
                                        height="40px"
                                        src="./header-logo.png"
                                        width="40px"
                              />
                    </WithStyles(Typography)>
          </WithStyles(Toolbar)>
</WithStyles(AppBar)>,
        "style": Object {
          "backgroundColor": "#000080",
          "position": "fixed",
          "top": 0,
          "width": "100%",
          "zIndex": 100,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Toolbar)>
            <WithStyles(Typography)
                        color="inherit"
                        variant="title"
            >
                        <img
                                    alt="logo"
                                    height="40px"
                                    src="./header-logo.png"
                                    width="40px"
                        />
            </WithStyles(Typography)>
</WithStyles(Toolbar)>,
          "position": "static",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Typography)
              color="inherit"
              variant="title"
>
              <img
                            alt="logo"
                            height="40px"
                            src="./header-logo.png"
                            width="40px"
              />
</WithStyles(Typography)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="logo"
                height="40px"
                src="./header-logo.png"
                width="40px"
/>,
              "color": "inherit",
              "variant": "title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "height": "40px",
                "src": "./header-logo.png",
                "width": "40px",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
